include/deal.II/numerics/rtree.h

```diff
@@ -0,0 +1,146 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_numerics_rtree_h
+#define dealii_numerics_rtree_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/point.h>
+
+#include <deal.II/boost_adaptors/bounding_box.h>
+#include <deal.II/boost_adaptors/point.h>
+#include <deal.II/boost_adaptors/segment.h>
+
+#include <boost/geometry.hpp>
+
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A wrapper for the boost::geometry library, used to compute an RTree from a
+ * collection of Point<dim> based spatial objects, BoundingBox<dim>, or
+ * Segment<dim> base spatial objects.
+ *
+ * An RTree may store any type of @p LeafType as long as it is possible to extract
+ * an @p Indexable that the RTree can handle and compare Values. The @p Indexable
+ * is a type adapted to Point, BoundingBox or Segment concept.
+ *
+ * In particular, given an Indexable (Point,  BoundingBox or Segment), @p Leaftype
+ * can by any of @p Indexable,  std::pair<Indexable, T>,
+ * boost::tuple<Indexable, ...> or std::tuple<Indexable, ...>.
+ *
+ * The optional argument @p IndexType is used only when adding elements to the
+ * tree one by one. If a range insertion is used, then the tree is built using
+ * the packing algorithm. Linear, quadratic, and rstar algorithms are available
+ * if one wants to construct the tree sequentially, however none of these is
+ * very efficient, and users should use the packing algorithm when possible,
+ * which construct the tree all at once, when you have all the leafs at your
+ * disposal. In this case, the @p IndexType parameter is not used, and you can
```

Comment:
This is a long sentence that's a bit hard to follow. May I suggest this minor reformulation (note the spelling corrections too):

Linear, quadratic, and rstar algorithms are available if one wants to construct the tree sequentially. 
However, none of these is very efficient, and users should use the packing algorithm when possible.
The packing algorithm constructs the tree all at once, and may be used when you have all the leaves at your disposal.